ydb/core/viewer/json_whoami.h

```diff
@@ -28,18 +29,48 @@ class TJsonWhoAmI : public TActorBootstrapped<TJsonWhoAmI> {
     {}
 
     void Bootstrap(const TActorContext& ctx) {
-        const auto& params(Event->Get()->Request.GetParams());
-        JsonSettings.EnumAsNumbers = !FromStringWithDefault<bool>(params.Get("enums"), false);
-        JsonSettings.UI64AsString = !FromStringWithDefault<bool>(params.Get("ui64"), false);
         ReplyAndDie(ctx);
     }
 
+    bool CheckGroupMembership(std::unique_ptr<NACLib::TUserToken>& token, const NProtoBuf::RepeatedPtrField<TString>& sids) {
+        if (sids.empty()) {
+            return true;
+        }
+        for (const auto& sid : sids) {
+            if (token->IsExist(sid)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     void ReplyAndDie(const  TActorContext &ctx) {
         NACLibProto::TUserToken userToken;
         Y_PROTOBUF_SUPPRESS_NODISCARD userToken.ParseFromString(Event->Get()->UserToken);
-        TStringStream json;
-        TProtoToJson::ProtoToJson(json, userToken, JsonSettings);
-        ctx.Send(Event->Sender, new NMon::TEvHttpInfoRes(Viewer->GetHTTPOKJSON(Event->Get()) + json.Str(), 0, NMon::IEvHttpInfoRes::EContentType::Custom));
+        NJson::TJsonValue json(NJson::JSON_MAP);
+        if (userToken.HasUserSID()) {
+            json["UserSID"] = userToken.GetUserSID();
+        }
+        if (userToken.HasGroupSIDs() && userToken.GetGroupSIDs().BucketsSize() > 0) {
+            NJson::TJsonValue& groupSIDs(json["GroupSIDs"]);
+            groupSIDs.SetType(NJson::JSON_ARRAY);
+            for (const auto& buckets : userToken.GetGroupSIDs().GetBuckets()) {
+                for (const auto& group : buckets.GetValues()) {
+                    groupSIDs.AppendValue(group);
+                }
+            }
+        }
+        if (userToken.HasOriginalUserToken()) {
+            json["OriginalUserToken"] = userToken.HasOriginalUserToken();
+        }
+        if (userToken.HasAuthType()) {
+            json["AuthType"] = userToken.GetAuthType();
+        }
+        auto token = std::make_unique<NACLib::TUserToken>(userToken);
+        json["IsViewerAllowed"] = CheckGroupMembership(token, AppData()->DomainsConfig.GetSecurityConfig().GetViewerAllowedSIDs());
+        json["IsMonitoringAllowed"] = CheckGroupMembership(token, AppData()->DomainsConfig.GetSecurityConfig().GetMonitoringAllowedSIDs());
+        json["IsAdministrationAllowed"] = CheckGroupMembership(token, AppData()->DomainsConfig.GetSecurityConfig().GetAdministrationAllowedSIDs());
```

Comment:
Could you please add these params somewhere to proto files? It's easier to look at protos, than to look how it's done in code

Reply:
Update: not proto, but swagger